hypertrace-trace-enricher/hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/clients/DefaultClientRegistry.java

```diff
@@ -165,4 +195,44 @@ protected Channel buildChannel(String host, int port) {
   protected Channel buildChannel(String host, int port, GrpcChannelConfig grpcChannelConfig) {
     return this.grpcChannelRegistry.forPlaintextAddress(host, port, grpcChannelConfig);
   }
+
+  private static Optional<KafkaLiveEventListener<EntityChangeEventKey, EntityChangeEventValue>>
+      getEntityChangeEventConsumer(
+          Config clientsConfig, BiConsumer<EntityChangeEventKey, EntityChangeEventValue> callback) {
+    if (clientsConfig.hasPath(ENTITY_CHANGE_EVENTS_CONSUMER_ENABLED_KEY)
+        && clientsConfig.getBoolean(ENTITY_CHANGE_EVENTS_CONSUMER_ENABLED_KEY)) {
+      String consumerName = clientsConfig.getString(ENTITY_CHANGE_EVENTS_CONSUMER_NAME_KEY);
+      Map<String, Object> serdeConfig =
+          Collections.singletonMap(
+              "schema.registry.url",
+              clientsConfig.getString(ENTITY_CHANGE_EVENTS_SCHEMA_REGISTRY_URL_KEY));
+      return Optional.of(
+          new KafkaLiveEventListener.Builder<EntityChangeEventKey, EntityChangeEventValue>()
+              .registerCallback(callback)
+              .build(
+                  consumerName,
+                  clientsConfig.getConfig(ENTITY_CHANGE_EVENTS_CONFIG_KEY),
+                  getEntityChangeEventKeySerde(serdeConfig),
+                  getEntityChangeEventValueSerde(serdeConfig)));
+    }
+    return Optional.empty();
+  }
+
+  private static Deserializer<EntityChangeEventKey> getEntityChangeEventKeySerde(
```

Comment:
naming - 'serde' refers to serializer/deserializer, when we lump both together. Given you're only working with deserialization here the naming should be adjusted.

Reply:
nit - method name too